keras_hub/src/models/gemma/gemma_attention.py

```diff
@@ -103,6 +105,18 @@ def _apply_rope(self, x, start_index):
         )
         return x
 
+    def _can_use_flash_attention(self):
+        if not has_flash_attention_support():
+            return False
+        if self.dropout > 0.0:
+            return False
+        if not running_on_tpu():
+            return False
+        sig = inspect.signature(ops.dot_product_attention)
+        if "attn_logits_soft_cap" not in sig.parameters:
```

Comment:
leave a comment explainer. e.g. soft cap is only supported on tpu and for recent versions of keras